src/test/java/org/elasticsearch/index/query/MatchAllQueryBuilderTest.java

```diff
@@ -0,0 +1,174 @@
+import org.elasticsearch.threadpool.ThreadPoolModule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.*;
+
+public class MatchAllQueryBuilderTest extends ElasticsearchTestCase {
+
+    private static final String MATCH_ALL_BOOST_1_5 = "{\"match_all\":{\"boost\":1.5}}";
+
+    protected QueryParseContext context;
+
+    protected Injector injector;
+
+    private XContentParser parser;
+
+    private MatchAllQueryBuilder testQuery;
+
+    @Before
+    public void setup() throws IOException {
+        Settings settings = ImmutableSettings.settingsBuilder()
+                .put("path.conf", this.getResourcePath("config"))
+                .put("name", getClass().getName())
+                .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+                .build();
+
+        Index index = new Index("test");
+        injector = new ModulesBuilder().add(
+                new EnvironmentModule(new Environment(settings)),
+                new SettingsModule(settings),
+                new ThreadPoolModule(settings),
+                new IndicesQueriesModule(),
+                new ScriptModule(settings),
+                new IndexSettingsModule(index, settings),
+                new IndexCacheModule(settings),
+                new AnalysisModule(settings),
+                new SimilarityModule(settings),
+                new IndexNameModule(index),
+                new IndexQueryParserModule(settings),
+                new FunctionScoreModule(),
+                new AbstractModule() {
+                    @Override
+                    protected void configure() {
+                        bind(ClusterService.class).toProvider(Providers.of((ClusterService) null));
+                        bind(CircuitBreakerService.class).to(NoneCircuitBreakerService.class);
+                    }
+                }
+        ).createInjector();
+
+        IndexQueryParserService queryParserService = injector.getInstance(IndexQueryParserService.class);
+        context = new QueryParseContext(index, queryParserService);
+
+        testQuery = createTestQuery();
+        String contentString = createXContent(testQuery).string();
+        parser = XContentFactory.xContent(contentString).createParser(contentString);
+    }
+
+    XContentBuilder createXContent(BaseQueryBuilder query) throws IOException {
+        XContentBuilder content = XContentFactory.jsonBuilder();
+        query.toXContent(content, null);
+        return content;
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        terminate(injector.getInstance(ThreadPool.class));
+    }
+
+    @Test
+    public void testToXContent() throws IOException {
+        XContentBuilder content = createXContent(new MatchAllQueryBuilder().boost(1.5f));
+        assertEquals(content.string(), MATCH_ALL_BOOST_1_5);
```

Comment:
I would love to find a way to test this without having to compare strings, the reason being that a single whitespace breaks things. We could do this by introducing json libraries that do that looking at json content, but I wouldn't go there. How about creating our `QueryBuilder` as we already do in this test, calling `toXContent` on it, and then parsing its result back via `fromXContent` and comparing that the obtained QueryBuilder is equal to the original one?
I know this approach would mix up toXContent and fromXContext which is not very unit test friendly, on the other hand we would make sure that they work well together. With more complex queries we could for instance randomize the options that we set on them and just verify that given a certain input the output is the same. Thoughts?